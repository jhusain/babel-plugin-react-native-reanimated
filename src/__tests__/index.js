import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

const projectRoot = path.join(__dirname, '../../')

expect.addSnapshotSerializer({
  print(val) {
    return val.split(projectRoot).join('<PROJECT_ROOT>/')
  },
  test(val) {
    return typeof val === 'string'
  },
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {filename: __filename, parserOpts: {plugins: ['jsx']}},
  tests: [
    {
      title: 'handles if statements',
      code: `
        const animate = require('../macro');

        const offsetX = animate(() => {
          if (offsetX === 30) {
            return 22;
          }
          else if (offsetY === 90) {
            offsetY = 55;
          }
          return 88;
        })
      `,
    },
  ],
})
