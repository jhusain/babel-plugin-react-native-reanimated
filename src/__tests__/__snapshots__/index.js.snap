// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros handles if statements: handles if statements 1`] = `

const animate = require('../macro');

const offsetX = animate(() => {
  if (offsetX === 30) {
    return 22;
  }
  else if (offsetY === 90) {
    offsetY = 55;
  }
  return 88;
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { Animated as _Animated, cond as _cond, set as _set, defined as _defined, not as _not, and as _and, or as _or, sub as _sub, pow as _pow, multiply as _multiply, divide as _divide, lessThan as _lessThan, modulo as _modulo, greaterThan as _greaterThan, lessOrEq as _lessOrEq, greaterOrEq as _greaterOrEq, neq as _neq, add as _add, eq as _eq, block as _block } from 'react-native-reanimated';


const offsetX = animate(() => {
  if (offsetX === 30) {
    return 22;
  } else if (offsetY === 90) {
    offsetY = 55;
  }
  return 88;
});

`;
